tests/std/tests/GH_004930_char_traits_user_specialization/test.cpp

```diff
@@ -0,0 +1,210 @@
+#endif // _HAS_CXX20
+        {
+            memmove(s, p, n);
+        }
+        return s;
+    }
+
+    static CONSTEXPR20 char_type* copy(char_type* s, const char_type* p, const size_t n) noexcept {
+#if _HAS_CXX20
+        if (is_constant_evaluated()) {
+            for (size_t i = 0; i != n; ++i) {
+                s[i] = p[i];
+            }
+        } else
+#endif // _HAS_CXX20
+        {
+            memmove(s, p, n);
+        }
+        return s;
+    }
+
+    static constexpr void assign(char_type& r, const char_type& d) noexcept {
+        r = d;
+    }
+    static CONSTEXPR20 char_type* assign(char_type* const s, const size_t n, const char_type c) {
+#if _HAS_CXX20
+        if (is_constant_evaluated()) {
+            for (size_t i = 0; i != n; ++i) {
+                s[i] = c;
+            }
+        } else
+#endif // _HAS_CXX20
+        {
+            memset(s, static_cast<unsigned char>(c), n);
+        }
+        return s;
+    }
+
+    static constexpr bool not_eof(int) noexcept {
+        return true;
+    }
+
+    static constexpr char_type to_char_type(const int_type i) noexcept {
+        return char_type(static_cast<unsigned char>(i));
+    }
+
+    static constexpr int_type to_int_type(const char_type i) noexcept {
+        return static_cast<unsigned char>(i);
+    }
+
+    static constexpr bool eq_int_type(const int_type c, const int_type d) noexcept {
+        if (c == -1) {
+            return d == -1;
+        } else {
+            return ((c ^ d) & odd_mask) == 0;
+        }
+    }
+
+    static constexpr int_type eof() noexcept {
+        return -1;
+    }
+};
+
+CONSTEXPR20 bool test_gh_4930() {
+    constexpr odd_char s_init[]{static_cast<odd_char>(0x55), static_cast<odd_char>(0x44), static_cast<odd_char>(0x33),
+        static_cast<odd_char>(0x22), static_cast<odd_char>(0x11), static_cast<odd_char>(0)};
+    constexpr odd_char s2_init[]{static_cast<odd_char>(0x83), static_cast<odd_char>(0x12), static_cast<odd_char>(0)};
+
+    using odd_string = basic_string<odd_char>;
+    {
+        odd_string s(s_init);
+
+        assert(s.length() == 5);
+        assert(s.find(static_cast<odd_char>(0x54)) == 1);
+        assert(s.find(static_cast<odd_char>(0x26)) == s.npos);
+
+        odd_string s2(s2_init);
+
+        assert(s.find_first_of(s2) == 2);
+        assert(s2.find_first_of(s) == 0);
+
+        assert(s.find_last_of(s2) == 3);
+        assert(s2.find_last_of(s) == 1);
+    }
+
+#if _HAS_CXX17
+    using odd_string_view = basic_string_view<odd_char>;
+    {
+        odd_string_view sv(s_init);
+
+        assert(sv.length() == 5);
+        assert(sv.find(static_cast<odd_char>(0x54)) == 1);
+        assert(sv.find(static_cast<odd_char>(0x26)) == sv.npos);
+
+        odd_string_view sv2(s2_init);
+
+        assert(sv.find_first_of(sv2) == 2);
+        assert(sv2.find_first_of(sv) == 0);
+
+        assert(sv.find_last_of(sv2) == 3);
+        assert(sv2.find_last_of(sv) == 1);
+    }
+#endif // _HAS_CXX17
+

```

Comment:
Apparently it is lacking lex compare coverage.

Reply:
I'm trying to expand the coverage now.

Reply:
Actually I don't even know whar that regex part does, not sure if it is lex compare

Reply:
I'm fine with this level of test coverage.